Sources/Nimble/Matchers/Equal.swift

```diff
@@ -50,47 +54,21 @@ public func equal<T: Comparable>(_ expectedValue: Set<T>?) -> Predicate<Set<T>>
 
 private func equal<T>(_ expectedValue: Set<T>?, stringify: @escaping (Set<T>?) -> String) -> Predicate<Set<T>> {
     return Predicate { actualExpression in
-        var errorMessage: ExpectationMessage =
-            .expectedActualValueTo("equal <\(stringify(expectedValue))>")
-
-        guard let expectedValue = expectedValue else {
```

Comment:
Why remove those?

Reply:
IMHO with Difference it looks better. Take a look at the screenshot in Jira. There are 4 cases: left nil, right nil, missing and extras.

Reply:
does it work when both inputs are nil?

Reply:
Yes, because it's a match. (not just saying, tested as well)